index.js

```diff
@@ -1133,6 +1136,8 @@
   //.
   //. The Either type satisfies the [Semigroup][], [Monad][], and [Extend][]
   //. specifications.
+  //.
+  //. This type is exported as `S.EitherType` for use with [sanctuary-def][].
```

Comment:
I think the two paragraphs we currently have should actually live until the `### Either type` heading:

```javascript
  //. ### Either type
  //.
  //. The Either type represents values with two possibilities: a value of type
  //. `Either a b` is either a Left whose value is of type `a` or a Right whose
  //. value is of type `b`.
  //.
  //. The Either type satisfies the [Semigroup][], [Monad][], and [Extend][]
  //. specifications.

  //# EitherType :: Type -> Type -> Type
  //.
  //. TK (should include a link to https://github.com/plaid/sanctuary-def#binarytype)

  //# Either :: TypeRep Either
  //.
  //. TK (should include a link to https://github.com/plaid/sanctuary#type-representatives)
```

What do you think of this structure?